engine/src/io/DataLoader.cpp

```diff
@@ -226,23 +224,42 @@ void data_loader::get_schema(Schema & schema, std::vector<std::pair<std::string,
 		schema.add_file(handle.uri.toString(true));
 	}
 
+	bool open_file = false;
+	while (this->provider->has_next()){
+		handles = this->provider->get_some(32, open_file);
```

Comment:
what is the logic behind getting 32 at a time?

Reply:
Honestly, I just picked a number. Any suggestions?

Reply:
I figure the number should be as high as can be without triggering issues. So to figure this out we need to see when the issues start occurring in different contexts. So to b eclear, the worst case is when 1 machine has the most amount of gpus. so with 8 or 16 you are generating 8 or 16 x the number of file handles. Set up something like 16 dask nodes on one machine then see when it starts crapping out. Pick some percentage of this as the top. I would say something like 10% of when it craps out. Or if I am going to rnadomly guess I would go for osmething like 10k 

This is of course in the case that its reading locally. If we are using  another file system like hdfs we need to see what error was happening.

Reply:
So, a starting point the default max number of file handles you can have open is 1024. So 1024/16 = 64.
In hdfs the default is 128 per process
So, given some preliminary research i can set it to 64 instead of 32